SummerSchool/app/src/main/java/nl/rug/www/rugsummerschools/controller/MainActivity.java

```diff
@@ -247,34 +253,58 @@ public void onComplete(@NonNull Task<AuthResult> task) {
                             // Sign in success, update UI with the signed-in user's information
                             Log.d(TAG, "signInWithCredential:success");
                             FirebaseUser user = mAuth.getCurrentUser();
-                            updateUI(user);
+                            handleProfilePicture(token, user);
+                            updateUI();
                         } else {
                             // If sign in fails, display a message to the user.
                             Log.w(TAG, "signInWithCredential:failure", task.getException());
                             Toast.makeText(MainActivity.this, "Authentication failed.",
                                     Toast.LENGTH_SHORT).show();
-                            updateUI(null);
+                            updateUI();
                         }
-
-                        // ...
+                        hideProgressDialog();
                     }
                 });
     }
 
+    private void handleProfilePicture(AccessToken token, final FirebaseUser user) {
+        GraphRequest request = GraphRequest.newMeRequest(token, new GraphRequest.GraphJSONObjectCallback() {
+            @Override
+            public void onCompleted(JSONObject object, GraphResponse response) {
+                Log.d(TAG, object.toString());
+                Log.d(TAG, response.toString());
+
+                Uri profilePicture;
+                try {
+                    profilePicture = Uri.parse("https://graph.facebook.com/" + object.getString("id") + "/picture?width=500&height=500");
```

Comment:
a bit too many constants hardcoded in this string...

Reply:
Right, I will modified this as constant string instead of hardcoding.